src/system/PredefinedProcedure.cpp

```diff
@@ -8,17 +8,36 @@
 #include <string>
 #include <utility>
 #include <vector>
+#include <iostream>
 
 using std::make_unique;
 using std::pair;
 using std::string;
 using std::vector;
+using std::cout;
 
 PredefinedProcedure::PredefinedProcedure(ProcKind kind, const string &name,
                                          const vector<pair<TypeNode *, bool>> &pairs, bool varargs, TypeNode *ret) :
-        ProcedureNode(make_unique<IdentDef>(name), nullptr), types_(), kind_(kind) {
+        ProcedureNode(make_unique<IdentDef>(name), nullptr), types_(), castIdx_(), kind_(kind) {
     auto type = overload(pairs, varargs, ret);
     this->setType(type);
+    castIdx_ = 0;
+    if (ret != nullptr) {
+        if (ret->kind() == TypeKind::TYPE) {
```

Comment:
These two `if` statements can be combined: `if (ret && ret->kind() == TypeKind::TYPE)`.

Reply:
Yes. Fixed.